cpp/src/parquet/arrow/arrow-schema-test.cc

```diff
@@ -849,5 +851,21 @@ TEST(InvalidSchema, ParquetNegativeDecimalScale) {
                 ToParquetSchema(arrow_schema.get(), *properties.get(), &result_schema));
 }
 
+TEST(TestFromParquetSchema, CorruptMetadata) {
+  // PARQUET-1565: ensure that an IOError is returned when the parquet file contains
+  // corrupted metadata.
+  auto path = test::get_data_file("PARQUET-1481.parquet",  /*is_good=*/ false);
+
+  try {
+    std::unique_ptr<parquet::ParquetFileReader> reader =
+        parquet::ParquetFileReader::OpenFile(path);
+    const auto parquet_schema = reader->metadata()->schema();
+    std::shared_ptr<::arrow::Schema> arrow_schema;
+    ASSERT_RAISES(IOError, FromParquetSchema(parquet_schema, &arrow_schema));
+  } catch (const ParquetException& e) {
+    FAIL() << "Exception thrown while reading file: " << e.what();
```

Comment:
I'm not sure this is necessary. AFAIK, gtest will catch exceptions and report them as errors.

Reply:
Yes good point, the try catch is unnecessary so I've deleted it.